test/CP/valueselection/learning/supervisedlearnedheuristic.jl

```diff
@@ -0,0 +1,501 @@
+                SeaPearl.GraphConv(3 => 64, Flux.leakyrelu)
+            ),
+            nodeChain=Flux.Chain(
+                Flux.Dense(64, 16, Flux.leakyrelu)
+            ),
+            outputChain=Flux.Dense(16, 4),
+        )
+
+        agent = RL.Agent(
+            policy=RL.QBasedPolicy(
+                learner=RL.DQNLearner(
+                    approximator=RL.NeuralNetworkApproximator(
+                        model=approximator_model,
+                        optimizer=ADAM(0.001f0)
+                    ),
+                    target_approximator=RL.NeuralNetworkApproximator(
+                        model=target_approximator_model,
+                        optimizer=ADAM(0.001f0)
+                    ),
+                    loss_func=Flux.Losses.huber_loss
+                ),
+                explorer=RL.EpsilonGreedyExplorer(
+                    ϵ_stable=0.01,
+                    rng=MersenneTwister(33)
+                )
+            ),
+            trajectory=RL.CircularArraySLARTTrajectory(
+                capacity=500,
+                state=SeaPearl.DefaultTrajectoryState[] => (),
+                action=Int => (),
+                legal_actions_mask=Vector{Bool} => (4,),
+            )
+        )
+
+        #EPISODE 1
+        trailer = SeaPearl.Trailer()
+        model = SeaPearl.CPModel(trailer)
+
+        x1 = SeaPearl.IntVar(1, 2, "x1", trailer)
+        x2 = SeaPearl.IntVar(1, 2, "x2", trailer)
+        x3 = SeaPearl.IntVar(2, 3, "x3", trailer)
+        x4 = SeaPearl.IntVar(1, 4, "x4", trailer)
+        SeaPearl.addVariable!(model, x1)
+        SeaPearl.addVariable!(model, x2)
+        SeaPearl.addVariable!(model, x3)
+        SeaPearl.addVariable!(model, x4)
+
+        SeaPearl.addConstraint!(model, SeaPearl.NotEqual(x1, x2, trailer))
+        SeaPearl.addConstraint!(model, SeaPearl.NotEqual(x2, x3, trailer))
+        SeaPearl.addConstraint!(model, SeaPearl.NotEqual(x3, x4, trailer))
+
+        variable_heuristic = SeaPearl.MinDomainVariableSelection()
+        lh = SeaPearl.SupervisedLearnedHeuristic(
+            agent,
+            eta_init=1.0,
+            eta_stable=0.0,
+            warmup_steps=2,
+            decay_steps=2
+        )
+        SeaPearl.update_with_cpmodel!(lh, model)
+
+        # At the beginning there is no helpSolution
+        @test isnothing(lh.helpSolution)
+
+        # Since eta_init = 1.0 an helpSolution is calculated
+        lh(SeaPearl.InitializingPhase, model)
+        @test !isnothing(lh.helpSolution)
+        @test lh.helpSolution == Dict{String,Union{Bool,Int64,Set{Int64}}}("x1" => 2, "x4" => 4, "x2" => 1, "x3" => 3)
+
+        # We check that the values returned by the heuristic correspond to those of the helpSolution.
+        # Variable 1
+        lh(SeaPearl.StepPhase, model, :Feasible)
+        x = variable_heuristic(model)
+        v = lh(SeaPearl.DecisionPhase, model, x)
+        @test v == lh.helpSolution[x.id]
+        lh(SeaPearl.EndingPhase, model, :Feasible)
```

Comment:
Is this line usefull?

Reply:
We have tried to keep the heuristic calls as close as possible, in order to detect possible problems in the future. However, I don't think that it will influence the result of the test in the current state of the removed. 

I am however in favor of keeping this line in the test.